code/modules/surgery/organs/liver.dm

```diff
@@ -38,12 +38,10 @@
 					//handle liver toxin filtration
 					var/toxamount
 					for(var/toxin in toxinList)
-						var/datum/reagent/toxin/T = toxin
-						if(istype(T))
-							toxamount += C.reagents.get_reagent_amount(T.type)
+						toxamount += C.reagents.get_reagent_amount(toxin.type)

```

Comment:
This means the total toxin damage a reagent will deal scales quadratically with amount injected.  Is this intentional?